pilot/adapter/config/ingress/conversion.go

```diff
@@ -93,15 +93,14 @@ func createIngressRule(name, host, path, domainSuffix string,
 	}
 
 	if path != "" {
-		if isRegularExpression(path) {
-			if strings.HasSuffix(path, ".*") && !isRegularExpression(strings.TrimSuffix(path, ".*")) {
-				rule.Match.Request.Headers[model.HeaderURI] = &routing.StringMatch{
-					MatchType: &routing.StringMatch_Prefix{Prefix: strings.TrimSuffix(path, ".*")},
-				}
-			} else {
-				rule.Match.Request.Headers[model.HeaderURI] = &routing.StringMatch{
-					MatchType: &routing.StringMatch_Regex{Regex: path},
-				}
+		if strings.HasSuffix(path, ".*") {
```

Comment:
I'd delete this one and be clear we don't do regex and don't pretend to anymore

Reply:
We do support regex. The problem is how to do both prefix and regex given a random string in the path.

Reply:
We can't just delete it without deprecation, but I'm +1 on deprecating the regex form of prefix matching, and eventually remove it.

One way to add regex support is to use an explicit tag ( instead of Path use Regex), or use 
a prefix 
Path: 
  regex:/.*[a-z]

Reply:
I'm ok with it, but we should be careful to document that `/a.b/.*` does not match `/axb/`.